### DISCUSSION TITLE
Creating an executable with `run` as a command

### DISCUSSION THREAD

**Discussion by KieranPringle** [erikhuda/thor#836] [Ideas]
I'm sure the title gives most of this away, but I've run into the following error and am looking for some way around it:

```
"run" is a Thor reserved word and cannot be defined as command
```

For context, I'm creating a build tool for internal use which creates workflows from config and then runs them, and `run` is the perfect word for this command.

Is there any lower-level Thor API manipulation I could do to get around this limitation? Something that would allow me to declare and register a command called `run`, still use the option and description declaration (even if that is also done by lower level API access), but avoid declaring it as a method called `run` in my class that extends Thor?

**Comment by Freakwill** [comment]
> I'm sure the title gives most of this away, but I've run into the following error and am looking for some way around it:
> 
> ```
> "run" is a Thor reserved word and cannot be defined as command
> ```
> For context, I'm creating a build tool for internal use which creates workflows from config and then runs them, and `run` is the perfect word for this command.
> 
> Is there any lower-level Thor API manipulation I could do to get around this limitation? Something that would allow me to declare and register a command called `run`, still use the option and description declaration (even if that is also done by lower level API access), but avoid declaring it as a method called `run` in my class that extends Thor?

See [my issue](https://github.com/erikhuda/thor/issues/647)
I thought we could implement a special method that need not call it explicitly in command line. It could be `run` or other proper words. I think it is also suitable for you.

**Comment by mddelk** [comment]
:+1:  for this feature.

Easy enough to rename the library `run` into `_run`(just a suggestion). The same approach could easily let us use `shell` as a command as well.
